autoload/codefmt.vim

```diff
@@ -156,7 +156,67 @@ if !exists('s:gofmt')
   call codefmtlib#AddDefaultFormatter(s:gofmt)
 endif
 
+" Formatter: autopep8
+if !exists('s:autopep8')
+  let s:autopep8 = {
+      \ 'name': 'autopep8',
+      \ 'setup_instructions': 'Install autopep8 and '}
+
+  function s:autopep8.IsAvailable() abort
+    return executable(s:plugin.Flag('autopep8_executable'))
+  endfunction
+
+  function s:autopep8.AppliesToBuffer() abort
+    return &filetype is# 'python'
+  endfunction
+
+
+  ""
+  " Reformat the current buffer with autopep8 or the binary named in
+  " @flag(autopep8_executable), only targeting the range between {startline} and
+  " {endline}.
+  function s:autopep8.FormatRange(startline, endline) abort
+    " Hack range formatting by formatting range individually, ignoring context.
+    let l:cmd = [ s:plugin.Flag('autopep8_executable'), "-"  ]
+    call maktaba#ensure#IsNumber(a:startline)
+    call maktaba#ensure#IsNumber(a:endline)
+    let l:lines = getline(1, line('$'))
+    let l:input = join(l:lines[a:startline - 1 : a:endline - 1], "\n")
```

Comment:
autopep8 1.0 and later support `--range line line`, which will give much better results than this hack. Unfortunately, Ubuntu seems to be stuck on a version almost 2 years old (0.9.1) which doesn't support it, so I could see people being stuck on an old version without it. I just filed https://bugs.launchpad.net/ubuntu/+source/autopep8/+bug/1414258 asking Ubuntu to update.

Could you file a feature request along with these changes to support true range formatting, with this as a fallback? "autopep8 --version" runs pretty quick for me, so I don't think running that once per vim session to detect native range support would be a problem.